app/models/announcement_chat_notifier.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class AnnouncementChatNotifier
+  def call(announce)
+    path = Rails.application.routes.url_helpers.polymorphic_path(announce)
+    url = "https://bootcamp.fjord.jp#{path}"
+
+    ChatNotifier.message(
+      "お知らせ：「#{announce.title}」\r#{url}",
+      webhook_url: ENV['DISCORD_ALL_WEBHOOK_URL']
+    )
```

Comment:
> announcement_chat_notifier.rbのcallメソッドの内容をDiscordNotifier.withを使った処理に置き換えるということであっていますか...？

おっしゃる通りです✨！

以下は最初のレビューでお伝えすればよかった内容で、後出しとなっていることご容赦いただけたらと思います🙏

```suggestion
    DiscordNotifier.with(announce: announce).announced.notify_now
```
シンプルにこうがいいかな？と思いました。
`DiscordNotifier.with...`の書き方になっているのにも意図があると思ったので（ #5480 が参考になりそうです）、置き換えに留めておいた方がいいかなと思いました。

動作の確認方法についても #5480 でyumさんが書いてくださっている方法 が参考になりそうです。